core/node/fee_model/src/lib.rs

```diff
@@ -202,11 +202,22 @@ fn compute_batch_fee_model_input_v2(
             (l1_batch_overhead_per_gas.as_u64() as f64 * compute_overhead_part) as u64;
 
         // We sum up the minimal L2 gas price (i.e. the raw prover/compute cost of a single L2 gas) and the overhead for batch being closed.
-        minimal_l2_gas_price + gas_overhead_wei
+        let calculated_price = minimal_l2_gas_price + gas_overhead_wei;
```

Comment:
what I'd suggest is, that we don't do this change in this method (compute_batch_fee_model_input_v2) - but instead create a new method "clip_batch_fee_model", which accepts the PubdataIndependentBatchFeeModelInput and mutates it to "fit" the bootloader params.

this way - if we remove the bootloader limits, we have easier cleanup, and moreover you'll not need the methods like maximum_l2_gas_price - but you can define these constants directly within this "clip" method.

And we should clearly mention it in the code, that the side effect, is that the chain can keep going (at a loss) even in the worst gas conditions (where the alternative would be to stop accepting the transactions until the conditions improve).

Reply:
Separated and added comments